django/db/models/fields/related.py

```diff
@@ -1885,6 +1899,30 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required arg for %s in Django 2.0. "
+                "Set it to models.CASCADE if you want to maintain the current default behavior. "
+                "See https://docs.djangoproject.com/en/%s/ref/models/fields/"
+                "#django.db.models.ForeignKey.on_delete" % (
+                    self.__class__.__name__,
+                    get_docs_version(),
+                ),
+                RemovedInDjango20Warning, 2)
```

Comment:
stacklevel 2 isn't useful when the warning is raised for OneToOneField. Ideally we could have stacklevel=3 for that case to give `question = models.OneToOneField(Question)` instead of `super(OneToOneField, self).__init__(to, on_delete, to_field=to_field, **kwargs)`.